crates/pallet-domains/src/staking_epoch.rs

```diff
@@ -533,9 +557,16 @@ fn finalize_nominator_deposit<T: Config>(
     deposit: BalanceOf<T>,
     total_stake: &mut BalanceOf<T>,
     total_shares: &mut T::Share,
+    pending_deposit_freeze_id: &FungibleFreezeId<T>,
+    staked_freeze_id: &FungibleFreezeId<T>,
 ) -> Result<(), TransitionError> {
     // calculate the shares to be added to nominator
-    let share_per_ssc = Perbill::from_rational(*total_shares, T::Share::from(*total_stake));
+    let share_per_ssc = if total_shares.is_zero() {
+        Perbill::one()
```

Comment:
Is this for the first nominator of the pool? Some comments can be helpful to understand the context.